### PR TITLE
adding SQL query sniffer

### PR DESCRIPTION
**Pull request by hardik1347**
SQL Sniffer Rule:
This will help us detect any SQL query changes in the PR

### CODE REVIEW

File: src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package io.github.thewisenerd.linters.sidekt.rules
+
+import io.gitlab.arturbosch.detekt.api.*
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtNamedFunction
+
+class SQLQuerySniffer(config: Config): Rule(config) {
+
+    private val sniffSQLAnnotation = listOf("SqlUpdate", "SqlCreate")
```
Comment (rajeshdixitudaan) [inline src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt]: Also add Select and Batch as well.

File: src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                        message = "SQL Query Sniffed $annotationValue with value $sqlQuery"
+                    )
+                )
+
+            }
+        }
+    }
+
+    private fun String.preProcessAnnotation(queryType: String): String {
+        val queryTypeExtractor = """@$queryType""".toRegex()
```
Comment (rajeshdixitudaan) [inline src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt:56]: Can you please provide the linters comment format.?

**Review by rajeshdixitudaan — APPROVED** [review]

File: src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt (hunk 1 above)
Comment (rajeshdixitudaan) [reply src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt]: Actually there was an SD in warehouse due to full table scan while SELECT.

**Review by rajeshdixitudaan — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt (hunk 1 above)
Comment (hardik1347) [reply src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt]: Done

File: src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt (hunk 2 above)
Comment (hardik1347) [reply src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt:56]: @rajeshdixitudaan Did not understand the concern can you elaborate please

Comment (rajeshdixitudaan) [reply src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/SQLQuerySniffer.kt:56]: what will be the final linter comment? URL jayega usme

**Pull request merged**